packages/components/src/input-control/styles/input-control-styles.tsx

```diff
@@ -235,6 +244,7 @@ export const Input = styled.input< InputProps >`
 		${ disabledStyles }
 		${ fontSizeStyles }
 		${ sizeStyles }
+		${ customPaddings }
```

Comment:
For the sake of making the logic of this override more explicit, I wonder if we should move the application of the custom paddings directly into the `sizeStyles` function.

Alternatively, we could at least add an inline comment making the padding override more obvious?

Reply:
Yes, something like this maybe? 38069ad

The custom padding API (as currently in this PR) is designed to be separate from the sizing concerns, but depending on the kind of use cases we discover, it might eventually make sense to make it a part of the sizing concerns.